benchmarks/babelstream/src/babelStreamMainTest.cpp

```diff
@@ -312,136 +345,252 @@ void testKernels()
                 bufAccInputAPtr,
                 bufAccInputBPtr,
                 bufAccOutputCPtr,
-                static_cast<DataType>(valA));
+                static_cast<DataType>(valA),
+                static_cast<DataType>(valB));
         },
         "InitKernel");
 
-    // Test the copy-kernel. Copy A one by one to B.
-    measureKernelExec(
-        [&]() { alpaka::exec<Acc>(queue, workDivCopy, CopyKernel(), bufAccInputAPtr, bufAccInputBPtr); },
-        "CopyKernel");
+    if(kernelsToBeExecuted == KernelsToRun::All)
+    {
+        // Test the copy-kernel. Copy A one by one to C.
+        measureKernelExec(
+            [&]() { alpaka::exec<Acc>(queue, workDivCopy, CopyKernel(), bufAccInputAPtr, bufAccOutputCPtr); },
+            "CopyKernel");
 
-    // Test the scaling-kernel. Calculate B=scalar*A.
-    measureKernelExec(
-        [&]() { alpaka::exec<Acc>(queue, workDivMult, MultKernel(), bufAccInputAPtr, bufAccInputBPtr); },
-        "MultKernel");
+        // Test the scaling-kernel. Calculate B=scalar*C. Where C = A.
+        measureKernelExec(
+            [&]() { alpaka::exec<Acc>(queue, workDivMult, MultKernel(), bufAccOutputCPtr, bufAccInputBPtr); },
+            "MultKernel");
 
-    // Test the addition-kernel. Calculate C=A+B. Where B=scalar*A.
-    measureKernelExec(
-        [&]()
-        { alpaka::exec<Acc>(queue, workDivAdd, AddKernel(), bufAccInputAPtr, bufAccInputBPtr, bufAccOutputCPtr); },
-        "AddKernel");
+        // Test the addition-kernel. Calculate C=A+B. Where B=scalar*C or B=scalar*A.
+        measureKernelExec(
+            [&]()
+            { alpaka::exec<Acc>(queue, workDivAdd, AddKernel(), bufAccInputAPtr, bufAccInputBPtr, bufAccOutputCPtr); },
+            "AddKernel");
+    }
 
-    // Test the Triad-kernel. Calculate C=A+scalar*B where B=scalar*A.
-    measureKernelExec(
-        [&]()
-        { alpaka::exec<Acc>(queue, workDivTriad, TriadKernel(), bufAccInputAPtr, bufAccInputBPtr, bufAccOutputCPtr); },
-        "TriadKernel");
+    if(kernelsToBeExecuted == KernelsToRun::All || kernelsToBeExecuted == KernelsToRun::Triad)
```

Comment:
```suggestion
    else if(kernelsToBeExecuted == KernelsToRun::Triad)
```
following https://github.com/UoB-HPC/BabelStream/blob/2f00dfb7f8b7cfe8c53d20d5c770bccbf8673440/src/main.cpp#L532

Reply:
There is a code repetition in the original code, both `run_triad `and `run_all` calls the triad kernel. Here, i call the same code piece for both cases.
 https://github.com/alpaka-group/alpaka/blob/7a8b2050bad28f0c6624c0413f7793d5a871049b/benchmarks/babelstream/src/main.cpp#L108